pkgs/development/python-modules/torcheval/torchtnt-nightly.nix

```diff
@@ -0,0 +1,59 @@
+{
+  buildPythonPackage,
+  fetchPypi,
+
+  # dependencies
+  fsspec,
+  numpy,
+  psutil,
+  pyre-extensions,
+  setuptools,
+  tabulate,
+  tensorboard,
+  torch,
+  tqdm,
+  typing-extensions,
+}:
+let
+  pname = "torchtnt-nightly";
+  version = "2023.1.25";
+
+in
+buildPythonPackage {
+  inherit pname version;
+  pyproject = true;
+
+  # The GitHub repo has no tag, so we fallback to PyPI instead
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-eFouZgdVQaqrXIKLY7geLV8GEVPGu6IHKIPK6aLJH8w=";
+  };
+
+  # The pypi archive does not supply any requirements
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace-fail 'install_requires=read_requirements("requirements.txt"),' "" \
+      --replace-fail 'extras_require={"dev": read_requirements("dev-requirements.txt")},' ""
+  '';
+
+  # requirements.txt
```

Comment:
I don't think we need that comment. Maybe they move to pyproject in the future and then we would need to update this